### PR TITLE
Pr fixes

### PR DESCRIPTION
**Pull request by danamansana**
Addresses most of the points raised in the previous review:

- Rename the `MySQLDBManager` and `PSQLClient` to `EnvisionwareManager` and `SierraDbClient`
- Return integers instead of strings for `batch_size` methods
- Adds some extra fallbacks in the `PcReserveClass` so we don't have to filter the db results. I will try to collect some examples where I saw this happening. Should we also be logging some errors in this case?

Not implemented:
- Batching the Kinesis writes. This requires an update to NYPLRubyUtil which will be in an upcoming PR.
- Combining `get_resp` and `match_to_ids` in the batchers. As I think about this more, I am not sure I like this idea. The choices are either i) have a third method that calls `get_resp` and `match_to_ids`, but this introduces repetition (since it has to be implemented in both classes, and is identical). ii) Combine the two methods into one, but I don't like this since I think there are really two distinct tasks, and I'd like to keep each method simple. I think this might seem a little less strange if both classes inherited from some `AbstractBatcher` class with a method `get_and_match_resps` which called `get_resp` and then `match_to_ids`. Essentially this implementation does the same thing with composition instead of inheritance. But if you really don't like it I can rewrite to combine the methods.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -2,10 +2,10 @@
 
 require_relative 'lib/state_manager'
 require_relative 'lib/query_builder'
-require_relative 'lib/pg_manager'
+require_relative 'lib/sierra_db_manager'
```
Comment (nonword) [inline app.rb]: Is it a "manager" or a "client"? The class name has `Client`, which I prefer personally.

File: lib/pc_reserve.rb [hunk 2]
```diff
@@ -1,32 +1,35 @@
...
 
 class PcReserve
   attr_accessor :event, :data, :patron
 
   def initialize (data, sierra_batch, patron_batch)
     @data = data
     @barcode = @data["pcrUserID"]
-    @patron = patron_batch[@barcode]
-    @id = @patron["id"].value.to_s
+    @patron = patron_batch[@barcode] || SafeNavigationWrapper.new(nil)
```
Comment (nonword) [inline lib/pc_reserve.rb:11]: Can you share (privately) some patron barcodes that aren't matching? I think we need to investigate why that's happening as I don't think it should be happening. If it actually may legitimately happen, we should talk with stakeholders about whether to skip the record (`raise` an error here) or allow `NULL`s

**Comment by nonword** [comment]
On the subject of the Batcher, another way to define what we're calling "(sierra|patron)_batch" are "things that take an array of ids and return a hash relating those same ids to some more data" or just "id lookups". That we're using them to build larger id lookups in _batches_ doesn't concern them, so I'm tripping on the name. But that's a minor thing.

I think what I was originally interested in was finding a way to reduce the public interface of those id lookups (patron_batch, sierra_batch) to just the interface actually necessary for building that larger lookup. As it is, your `Batcher` class reaches into each of them and calls two methods: `get_resp` and `match_to_ids`. That makes it difficult to follow the logic because those two methods are related only in how they're called by an outside class. That's why I proposed adding a third method to bind them together (I proposed `get_id_matches`, but maybe just `lookup`?). That both PatronBatch and SierraBatch happen to implement `get_resp` and `match_to_ids` is an internal concern, not one of `Batcher`s; I was thinking that the only necessary property of an id lookup like PatronBatch and SierraBatch is that they implement a single essential method `get_id_matches`. I think the super class idea could work, but you're introducing a fourth class into the mix, so I do worry that will make it even harder to follow the logic and will constrain the internals of SierraBatch and PatronBatch, which doesn't seem necessary. So on balance - assuming you don't want to add a `get_id_matches` method to both *Batch classes (which happens to be implemented identically in both) - I think I withdraw the whole original complaint because I think it's easier to follow `Batcher` calling two methods on two similar classes than `Batcher` calling a method inherited by those classes, which calls on two other methods actually in those classes. Essentially, I worry we'd be introducing cognitive load for not much payoff

File: app.rb (hunk 1 above)
Comment (danamansana) [reply app.rb]: Ah good catch. Fixed

File: batch.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
...ey\":\"M/ApTJw2YpYkv/z2YIZTf3z4GJcHAy.\",\"minutes_used\":33,\"transaction_et\":\"2021-05-06 12:40:23 -0400\",\"branch\":\"PELHAM BAY\",\"area\":\"Internet/Full Service\",\"staff_override\":\"UNKNOWN\"},{\"patron_id\":null,\"ptype_code\":null,\"patron_home_library_code\":null,\"pcode3\":null,\"postal_code\":null,\"geoid\":null,\"key\":\"3oTLqkMlzsAdzz3Q.PfwnzWI5ypAVaW\",\"minutes_used\":11,\"transaction_et\":\"2021-05-06 12:21:04 -0400\",\"branch\":\"PELHAM BAY\",\"area\":\"Internet/Full Service\",\"staff_override\":\"UNKNOWN\"},{\"patron_id\":null,\"ptype_code\":null,\"patron_home_library_code\":null,\"pcode3\":null,\"postal_code\":null,\"geoid\":null,\"key\":\"q6DcA7kNRGjFsCHZ9wfreS0MmV/H9cC\",\"minutes_used\":21,\"transaction_et\":\"2021-05-06 12:33:36 -0400\",\"branch\":\"PELHAM BAY\",\"area\":\"Internet/Full Service\",\"staff_override\":\"UNKNOWN\"},{\"patron_id\":null,\"ptype_code\":null,\"patron_home_library_code\":null,\"pcode3\":null,\"postal_code\":null,\"geoid\":null,\"key\":\".6NwKBJQVXkcThr6mu6H7CqfuvqPBc2\",\"minutes_used\":11,\"transaction_et\":\"2021-05-06 12:30:08 -0400\",\"branch\":\"PELHAM BAY\",\"area\":\"Internet/Full Service\",\"staff_override\":\"UNKNOWN\"}]"
```
Comment (nonword) [inline batch.rb:4]: What is this?

**Review by nonword — APPROVED** [review]

**Pull request merged**